docs/DeveloperGuide.md

```diff
@@ -8,8 +8,237 @@
+- **Storage**: Responsible for loading and saving data from and to the disk.
+
+---
+
+### 1.2 UI Component
+
+The UI component is responsible for displaying information to the user and reading input. It interacts with the user by printing messages and prompts, and retrieves commands for the rest of the application to process.
+
+### 1.3 Logic Component
+
+The logic component is handled by the `Parser` and `Command` classes. The `Parser` interprets the user's input and returns a command object, which is then executed to perform the desired operation.
+
+### 1.4 Model Component
+
+The model component includes the `Internship`, `InternshipList`, `Deadline` and `DeadlineList` classes.
+
+`Internship`: Represents the individual internship entries.
+
+`InternshipList`: Manages a collection of Internship objects, providing methods for adding, retrieving, updating, sorting and removing internships.
+
+`Deadline`: Represents each individual deadline associated with an internship, holding data specific to that deadline (description, date).
+
+Each `Internship` object contains a `Deadline` list, and has methods to add, retrieve or remove deadlines associated with that internship.
+
+This is represented with the class diagram below:
+
+![](UML/Internship-Model_Component.png)
+
+### 1.5 Storage Component
+
+The storage component is responsible for saving the current state of the `InternshipList` to a file and loading it back when the application is restarted. This ensures that users' progress is preserved between sessions.
+The data from the internshipList is saved into a .txt file.
+
+There are two main functions, `loadFromFile` and `saveFromFile`.
+
+The following sequence diagrams depict how the Storage Functions work.
+![](UML/loadFromFile.png)
+![](UML/saveToFile.png)
+
+
+## 2. Implementation
+
+### 2.1 Parsing Input
+
+The `Parser` class is responsible for parsing the input strings provided by the user. It breaks down the input into individual components like the command and its accompanying arguments. After parsing, it returns a `Command` object.
+
+### 2.2 Command Execution
+
+The `Command` class and its subclasses handle the execution of specific commands. Once a command is parsed, it is executed with the required arguments, and the result is displayed to the user.
+
+### 2.3 Task Management
+
+The `InternshipList` class handles adding, removing, and retrieving internships from the list. It interacts with both the UI and the storage components to ensure that updates are reflected in the user interface and persisted to disk.
+
+---
+
+### Overview
+
+The `EasInternship` class serves as the entry point of the application. It manages the application's main loop, where the user is continually prompted for input, and commands are processed in response. The class is responsible for initializing the UI, loading saved data, and handling user input until the user chooses to exit the program.
+
```

Comment:
I feel that the overview would be more fitting being placed earlier in the documentation guide, perhaps you can give this some consideration?